ext/spl/spl_fixedarray.c

```diff
@@ -43,6 +43,8 @@ ZEND_GET_MODULE(spl_fixedarray)
 typedef struct _spl_fixedarray { /* {{{ */
 	zend_long size;
 	zval *elements;
+	unsigned int is_resizing:1;
```

Comment:
Compared to creating a temporary copy of zvals to destroy, this would also add another 4 or 8 bytes (with 64-bit struct field alignment?) to each SplFixedArray instance (and a tiny bit more time initializing it), when most applications wouldn't use setSize (according to other remarks it was meant more as an escape hatch than as a way to support vector-like functionality)

Some progress was made in reducing SplFixedArray's memory usage and instantiation time in https://github.com/php/php-src/pull/6552